### ISSUE TITLE
Authentication failing

### ISSUE THREAD

**Issue by soabwahott** [ndbeals/keep-exporter#26]
This has been broken for a couple of months, fails with this error:

Error: Invalid value: Password login failed: ('NeedsBrowser', 'To access your account, you must sign in on the web. Touch Next to start browser sign-in.')

**Comment by chaoscreater** [comment]
Same issue

**Comment by ndbeals** [comment]
Do you use 2FA for your google account?

**Comment by soabwahott** [comment]
Yes

**Comment by ndbeals** [comment]
Did you follow the section in the readme about how to authenticate with 2FA? https://github.com/ndbeals/keep-exporter#notes

I suspect the initial error message you got about using the browser is because of the 2FA sign-in flow.

**Comment by soabwahott** [comment]
Yes, I followed the instructions and setup an app password. This worked fine for several months and then all of a sudden stopped. I tried creating a new app password but this didn't help. I'm assuming Google changed something that broke the code.

**Comment by chaoscreater** [comment]
same for me. Was working for months, then it broke. My account already had 2FA configured and I was using app password. Nothing has changed. 

I've tried other projects similar to this as well and getting the same issue.

Also, if I input the wrong password intentionally, the error will show an error that specifically points out it's a wrong password. So it's not an issue with the app password generated by Google, it's an issue with the new authentication process in their code that probably broke the script.

**Comment by ndbeals** [comment]
I see, thanks for the further information, that will help me when I go to recreate this during bug fixing

**Comment by chaoscreater** [comment]
I got it working. It's not an issue with my Google app password.

I've had to remove all the virtualenv created by poetry and re-do everything from scratch. Tested working with Python 3.9. Seems like it's broken on 3.7.

**Comment by soabwahott** [comment]
thanks @chaoscreater, I did the same, deleted my virtualenv and recreated it. The code is now working with no issues.

I'm also using python 3.9. 

I guess we can close the issue!

**Comment by chaoscreater** [comment]
> thanks @chaoscreater, I did the same, deleted my virtualenv and recreated it. The code is now working with no issues.
> 
> I'm also using python 3.9.
> 
> I guess we can close the issue!

no problem. There are also these other projects I'm using as backup, in case the solution I currently have fails.

https://github.com/kiwiz/keep-cli
https://github.com/kiwiz/gkeepapi/issues/135
